db/seeds/scheme_ten.rb

```diff
@@ -50,18 +53,46 @@
 csv_file = File.open(file_path, 'r:ISO-8859-1')
 csv = CSV.parse(csv_file, headers: true)
 
+module OffenceCSVRowExtensions
+  def description
+    self['description'].strip
+  end
+
+  def offence_category
+    OffenceCategory.find_by(number: self['category'])
+  end
+
+  def offence_band
+    OffenceBand.find_by(offence_category: offence_category, description: self['band'])
+  end
+
+  def contrary_to
+    self['contrary_to']
+  end
+
+  def year_chapter
+    self['year_chapter']
+  end
+end
+
+class CSV::Row
+  include OffenceCSVRowExtensions
+end
+
 csv.each do |row|
-  # row desc : Category,Band,Offence,Contrary to,Year and Chapter
-  description = row[2].strip
-  offence_category = OffenceCategory.find_by(number: row[0])
-  offence_band = OffenceBand.find_by(offence_category: offence_category, description: row[1])
-  unique_code = OffenceCodeSeederSchemeTen.new(description, offence_band.number.to_s.each_byte.to_a.join(''), row[3]).unique_code
-  offence = Offence.find_or_create_by!(
-    offence_band: offence_band,
-    description: description,
-    unique_code: unique_code,
-    contrary: row[3],
-    year_chapter: row[4]
+  offence = SeedHelper.find_or_create_scheme_10_offence!(
+    offence_band: row.offence_band,
+    description: row.description,
+    contrary: row.contrary_to,
+    year_chapter: row.year_chapter
   )
   OffenceFeeScheme.find_or_create_by(offence: offence, fee_scheme: agfs_fee_scheme_ten)
-end
\ No newline at end of file
+end
+
+# regenerate unique codes based on offence description and band where order is significant
+require Rails.root.join('lib','data_migrator','offence_unique_code_migrator').to_s
```

Comment:
❓ What's the `.to_s` for in this **require** ? 😕

Reply:
copy-paste - i thought it was required but, after trying, it is not.